gws.tf

```diff
@@ -8,7 +8,7 @@ resource "aws_eip" "default" {
 }
 
 resource "aws_nat_gateway" "default" {
-  count         = "${length(var.availability_zones)}"
+  count         = "${var.nat_gateway_eip_enabled == true ? 0 : length(var.availability_zones)}"
```

Comment:
No, this needs to be a different flag (e.g. `nat_gateway_enabled`). Private subnets cannot egress to the internet without a NAT gateway (e.g. install packages). It'll be a rare situation where we'll disable a NAT gateway entirely.